### PR TITLE
Document jpaRest filters

### PR DESCRIPTION
**Pull request by daleelR**
Added documentation on how to use filters with jpaRest.
_Hint: click the paper icon at the top right of the readme.md file to see the diff in the formatted version._

### CODE REVIEW

**Review by JWeston-HO — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -74,3 +74,67 @@ is as follows
...
+#### Logical Operators
+When forming a filter expression using logical operators the following conditions must be met:
+- The left side operand must be a reference to a field on the entity. Literal values are not supported here.
+- The right side operand can be a field reference or a literal value, however the data type of the right  
+  side operand must match the data type of the left side operand.
+- The following operators are supported: `==, !=, >=, >, <=, <`
+- The operators can also be provided in the equivalent textual form such as: `eq, ne, ge, gt, le, lt`
+
+An example of the filter request parameter using a logical operator is as follows:
+`filter=id == 'dc27d5aa-4e7d-474e-98b6-ebf9aae9a471'`
```
Comment (holly-blues) [inline README.md:103]: Is the entity name case sensitive?

File: README.md [hunk 2]
```diff
@@ -74,3 +74,67 @@ is as follows
...
+`filter=in(id, '27e813a2-bb28-11ec-8422-0242ac120001', '27e813a2-bb28-11ec-8422-0242ac120003')`
+
+
+### Combining Conditions
+There are a few additional logical operators that can be applied and also used to build a filter expression made up of multiple conditions.
+- This can be applied with any of the following operators: `&&, ||, !`
+- The operators can also be provided in the equivalent textual form such as: `and, or, not`
+
+An example of the filter request parameter using additional logical operators is as follows:
+`filter=!in(id, '27e813a2-bb28-11ec-8422-0242ac120001', '27e813a2-bb28-11ec-8422-0242ac120003')`
```
Comment (holly-blues) [inline README.md:124]: Can you combine filters? E.g. startDate gt X and endDate lt Y

If so an example would be really helpful.

File: README.md [hunk 3]
```diff
@@ -74,3 +74,67 @@ is as follows
 ## Merging into Main
 Once a PR is merged into the main branch the snapshot suffix will be deleted from the version.
 The assigned version number will be used to create a version release in artifactory and a Github tag for backup.
+
+
+## Using Filters
+
+### Overview
+JpaRest supports the use of filtering data during retrieval for certain get endpoints. These filters should be constructed in accordance with Spring Expression Language (SpEL) as they will then be converted into predicates used to query the database.
```
Comment (laurieboyes) [inline README.md:82]: I like how you format GET below, can we make it consistent?
```suggestion
JpaRest supports the use of filtering data during retrieval for certain `GET` endpoints. These filters should be constructed in accordance with Spring Expression Language (SpEL) as they will then be converted into predicates used to query the database.
```

**Review by laurieboyes — APPROVED** [review]
Poetry

File: README.md (hunk 1 above)
Comment (daleelR) [reply README.md:107]: I've now updated the remaining areas to mention if they're case sensitive.

File: README.md (hunk 2 above)
Comment (daleelR) [reply README.md]: indeed you can. have now added an example

File: README.md (hunk 3 above)
Comment (daleelR) [reply README.md]: ah yes missed that one. updated.

**Review by holly-blues — APPROVED** [review]

**Review by laurieboyes — APPROVED** [review]

**Pull request merged**